lib/python/treadmill_aws/hostmanager.py

```diff
@@ -47,6 +51,9 @@ def create_host(ec2_conn, ipa_client, image_id, count, domain,
         host_ctx['otp'] = ipa_host['result']['result']['randompassword']
         user_data = render_manifest(host_ctx)
 
+        for hostgroup in hostgroups:
```

Comment:
do we abort on errors? if we delete the host, it will be removed from the groups automatically, correct?

Reply:
If IPA returns an error when adding the host to a hostgroup (say if the hostgroup did not exist), we do abort and return an exception. The OTP for that host would be lost and the IPA host record can be cleaned up later. 
Yes, hosts are automatically removed from hostgroups on deletion.